pkg/gardenlet/metrics/metrics.go

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	runtimemetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// Namespace is the metric namespace for the gardenlet.
+const Namespace = "gardenlet"
+
+var (
+	// Factory is used for registering metrics in the controller-runtime metrics registry.
+	Factory = promauto.With(runtimemetrics.Registry)
+	// ShootOperationTimings defines the histogram shoot_operation_duration_seconds.
+	ShootOperationTimings = Factory.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: Namespace,
+			Name:      "shoot_operation_duration_seconds",
```

Comment:
Can you add the `shoot_.+` metrics to the allowlist for collecting metrics in the performance prometheus in CI? [ref](https://github.com/gardener/gardener/blob/b6b857890afabfa10c75a2016f3f02ddde5762a6/pkg/provider-local/webhook/prometheus/mutator.go#L61)